packages/server/src/api/controllers/static/index.ts

```diff
@@ -287,6 +293,37 @@ export const getSignedUploadURL = async function (ctx: Ctx) {
     } catch (error: any) {
       ctx.throw(400, error)
     }
+  } else if (datasource?.source === "GOOGLE_CLOUD") {
+    const { bucket, key } = ctx.request.body || {}
+
+    const parsedKey = datasource?.config?.privateKey
+      .split(String.raw`\n`)
+      .join("\n")
+
+    publicUrl = `https://storage.cloud.google.com/${bucket}/${key}`
```

Comment:
I believe both `bucket` and `key` can be `undefined` here. We likely want to do a check before attempting to make any calls.